429 learningtesttool fails on debian10

popescu-v opened the pull request:
closes #429

popescu-v committed:
Fix typo in comment

popescu-v committed:
Remove duplicate code: the exact same lines are present before the removed lines

marcboulle reviewed (approved):
LGTM
(un petit commentaire à préciser)

popescu-v committed:
Add the `--allow-run-as-root` option to the OpenMPI executable

Thus, OpenMPI is instructed to allow to be run as root in a portable
way.

The `OMPI_ALLOW_RUN_AS_ROOT` and `OMPI_ALLOW_RUN_AS_ROOT_CONFIRM`
environment variables are not taken into account by `mpiexec` on Debian
10, because they were not supported at the time Debian 10 was released.
